cabal-install/tests/PackageTests/InitConfig/Check.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module PackageTests.InitConfig.Check
```

Comment:
Optional: could this be written as a unit test instead? Package tests are really slow (because we need to actually compile code) and I'm trying to avoid them. You can still do I/O in a unit test. In this case you could have the test call `createDefaultConfigFile` or `initConfigAction` directly.

You can use `Distribution.Simple.Utils.withTempDir` to create a temporary directory instead of using `packageTestsDirectory`. Using a non-temp dir risks issues e.g. if we run tests in parallel. Example test:

```haskell
readSourcePackagesInDirTest :: Assertion
readSourcePackagesInDirTest = do
  sysTmpDir <- Directory.getTemporaryDirectory
  Utils.withTempDirectory silent sysTmpDir "cabal-test" $ \ tmpDir -> do
    let pkgDir = tmpDir </> "a"
    Directory.createDirectory pkgDir
    -- Write a trivial cabal file.
    writeFile (pkgDir </> "a.cabal") $ unlines [
      "name: a",
      "version: 1.2.3.4"
      ]
    index <- SourceTree.readSourcePackagesInDir silent tmpDir
    ...
    assertBool "Package should exist in index." $
      PackageIndex.elemByPackageId index pkgid
```